crates/components/poseidon-bn256-pad14/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+//! Poseidon hash over the bn256 curve with the input padding length of 14 field
+//! elements.
+
+use poseidon_circomlib::{hash as hash_inner, F};
+
+/// Maximum allowed bytelength of plaintext.
+const MAX_PLAINTEXT: usize = 434;
+/// How many bytes to pack into a single field element.
+const BYTES_PER_FIELD: usize = 31;
+/// The length to pad the plaintext field element count to.  
+const PAD_LENGTH: usize = 14;
+
+/// Hashes the given `plaintext` (padding it) and `salt`, returning the digest
+/// as bytes.
+///
+/// # Panics
+///
+/// Panics if the plaintext or salt lengths are not correct.
+pub fn hash(plaintext: &[u8], salt: &[u8]) -> Vec<u8> {
+    let mut out = hash_to_field(plaintext, salt).to_bytes();
+    out.reverse();
+    out.to_vec()
+}
+
+/// Hashes the given `plaintext` (padding it) and `salt`, returning the digest
+/// as a field element.
+///
+/// # Panics
+///
+/// Panics if the plaintext or salt lengths are not correct.
+pub fn hash_to_field(plaintext: &[u8], salt: &[u8]) -> F {
+    assert!(plaintext.len() <= MAX_PLAINTEXT);
+
+    let mut plaintext: Vec<F> = plaintext
+        .chunks(BYTES_PER_FIELD)
+        .map(bytes_to_f)
+        .collect::<Vec<_>>();
+
+    // Zero-pad if needed.
+    plaintext.extend(vec![F::zero(); PAD_LENGTH - plaintext.len()]);
+
+    plaintext.push(bytes_to_f(salt));
+
+    hash_inner(&plaintext)
+}
+
+/// Converts a little-endian byte representation of a scalar into a `F`.
```

Comment:
plaintext and salt passed into this crate are expected to be in little-endian format?

Reply:
and hash() is returning the result in big endian? (due to `reverse()`)

Reply:
Good catch! 
I didn't think I needed to write a test for something as simple as what this crate does. It would have caught this error. Now I will add a test.